samples/bluetooth/alexa_gadget/src/gadgets/CMakeLists.txt

```diff
@@ -0,0 +1,187 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+# The Gadgets Profile relies on the following external dependencies that
+# must be downloaded and run before the build stage:
+# - nanopb generator
+# - protocol buffer definitions
+
+# Note: as there doesn't seem to be an easy way of checking the version of the nanopb generator (protoc-gen-nanopb binary)
+# there is the possibility of a version mismatch between the generator and the nanopb source files that is used in the firmware sample.
+# If there is a mismatch, there will be a compilation error in the build step.
+# To fix such an error, one must either update NANOPB_REPO_TAG or download a generator version that matches NANOPB_REPO_TAG
+# (https://jpa.kapsi.fi/nanopb/download/)
+
+set(GADGETS_PROTO_REPO "https://github.com/alexa/Alexa-Gadgets-Embedded-Sample-Code.git")
+set(GADGETS_PROTO_TAG "face92d8c62184832793f518bb1f19379538c5c1")
+
+set(NANOPB_REPO "https://github.com/nanopb/nanopb.git")
+set(NANOPB_REPO_TAG "nanopb-0.4.1")
+
+# Check local folder for protoc and nanopb binaries presence
+file(GLOB_RECURSE LOCAL_PROTOC_BIN "${PROJECT_SOURCE_DIR}/*/generator-bin/protoc")
+if (NOT LOCAL_PROTOC_BIN)
+    file(GLOB_RECURSE LOCAL_PROTOC_BIN "${PROJECT_SOURCE_DIR}/*/generator-bin/protoc.exe")
+endif()
+file(GLOB_RECURSE LOCAL_NANOPB_BIN "${PROJECT_SOURCE_DIR}/*/generator-bin/protoc-gen-nanopb*")
+
+# Check system for protoc and nanopb binaries presence
+find_program(PROTOC_BIN protoc)
+find_program(NANOPB_BIN protoc-gen-nanopb)
+
+# Prefer local binaries to make version matching with nanopb source repo easier
+if (LOCAL_PROTOC_BIN)
+    set(PROTOC_BIN ${LOCAL_PROTOC_BIN})
+endif()
+
+if (LOCAL_NANOPB_BIN)
+    set(NANOPB_BIN ${LOCAL_NANOPB_BIN})
+endif()
+
+if ((NOT NANOPB_BIN) OR (NOT PROTOC_BIN))
+    message(STATUS "nanopb not found.")
+    message(STATUS "Please download ${NANOPB_REPO_TAG} from here https://jpa.kapsi.fi/nanopb/download/")
+    message(STATUS "and extract to either this sample directory or somewhere cmake can find it via system PATH.")
+    message(FATAL_ERROR "${NANOPB_REPO_TAG} required by this project.")
+endif()
+
+message(STATUS "Using protoc: ${PROTOC_BIN}")
+message(STATUS "Using protoc-gen-nanopb: ${NANOPB_BIN}")
+message(STATUS "Using nanopb source tag: ${NANOPB_REPO_TAG}")
+
+function(compile_protobufs_in_dir dir recursive)
+    # Find all .proto files and include directories
+    if (recursive)
+        file(GLOB_RECURSE PROTO_FILES "${dir}/*.proto")
+    else()
+        file(GLOB PROTO_FILES "${dir}/*.proto")
+    endif()
```

Comment:
How many files are we talking about ?
Globbing should be used with care:
https://cmake.org/cmake/help/v3.18/command/file.html#filesystem

Reply:
~30 files scattered throughout ~10 directories (https://github.com/alexa/Alexa-Gadgets-Embedded-Sample-Code/tree/master/AlexaGadgetsProtobuf)